src/execution/index/art/art.cpp

```diff
@@ -988,67 +978,160 @@ void ART::CheckConstraintsForChunk(DataChunk &input, ConflictManager &conflict_m
 //===--------------------------------------------------------------------===//
 
 BlockPointer ART::Serialize(MetaBlockWriter &writer) {
+
 	lock_guard<mutex> l(lock);
-	auto old_memory_size = memory_size;
 	if (tree) {
-		serialized_data_pointer = tree->Serialize(*this, writer);
+		serialized_data_pointer = tree.Serialize(*this, writer);
 	} else {
 		serialized_data_pointer = {(block_id_t)DConstants::INVALID_INDEX, (uint32_t)DConstants::INVALID_INDEX};
 	}
-	IncreaseAndVerifyMemorySize(old_memory_size);
+
+	UpdateMemoryUsage();
 	return serialized_data_pointer;
 }
 
+//===--------------------------------------------------------------------===//
+// Vacuum
+//===--------------------------------------------------------------------===//
+
+vector<bool> ART::InitializeVacuum(vector<FixedSizeAllocator *> &allocators) {
```

Comment:
Can we perhaps wrap the `vector<bool>` into a struct (`ARTVacuumFlags`) so it is more clear what it is?